src/AgentCommerce/Util/PayPalCartFactory.php

```diff
@@ -0,0 +1,91 @@
+<?php declare(strict_types=1);
+/*
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Swag\PayPal\AgentCommerce\Util;
+
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\PayPalSDK\Struct\AgenticCommerce\V1\Address;
+use Shopware\PayPalSDK\Struct\AgenticCommerce\V1\CartItemCollection;
+use Shopware\PayPalSDK\Struct\AgenticCommerce\V1\PayPalCart;
+use Swag\PayPal\AgentCommerce\Exception\AgentException;
+
+class PayPalCartFactory
+{
+    public function create(array $data): PayPalCart
+    {
+        $payPalCart = (new PayPalCart())->assign($data);
+
+        $this->validateCustomerData($payPalCart);
+
+        if (!$payPalCart->isset('items') || !$payPalCart->getItems()->count()) {
+            throw AgentException::requiredFieldsMissing('items');
+        }
+
+        $this->validateItems($payPalCart->getItems());
+
+        return $payPalCart;
+    }
+
+    private function validateCustomerData(PayPalCart $cart): void
+    {
+        $customer = $cart->getCustomer();
+        if (!$customer) {
+            return;
+        }
+
+        if (!$customer->getEmailAddress()) {
+            throw AgentException::requiredFieldsMissing('customer email');
```

Comment:
Regarding such exceptions, I would prefer path style variables for better visibility and identifying issues (with or without `cart.`):
```suggestion
            throw AgentException::requiredFieldsMissing('cart.customer.emailAddress');
```